GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/1
----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 1 processes
----------------------------------------------------------------------------------------------------
/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:604: UserWarning: Checkpoint directory /bme002.mnt/project/bme-lrossen-20192236/Master_Thesis/DiffSBDD/logs/SE3-cond-fullAtom/SE3-cond-fullAtom/checkpoints exists and is not empty.
  rank_zero_warn(f"Checkpoint directory {dirpath} exists and is not empty.")
Restoring states from the checkpoint path at /bme002.mnt/project/bme-lrossen-20192236/Master_Thesis/DiffSBDD/logs/SE3-cond-fullAtom/SE3-cond-fullAtom/checkpoints/last.ckpt
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1,2]
  | Name | Type            | Params
-----------------------------------------
0 | ddpm | ConditionalDDPM | 2.8 M
-----------------------------------------
2.8 M     Trainable params
501       Non-trainable params
2.8 M     Total params
11.048    Total estimated model params size (MB)
Restored all states from the checkpoint file at /bme002.mnt/project/bme-lrossen-20192236/Master_Thesis/DiffSBDD/logs/SE3-cond-fullAtom/SE3-cond-fullAtom/checkpoints/last.ckpt
SLURM auto-requeueing enabled. Setting signal handlers.
Clipped gradient with value 0.4 while allowed 0.4
Clipped gradient with value 0.6 while allowed 0.5
Clipped gradient with value 0.5 while allowed 0.4
Clipped gradient with value 0.5 while allowed 0.5
Clipped gradient with value 0.4 while allowed 0.4
Clipped gradient with value 0.7 while allowed 0.4
Clipped gradient with value 0.6 while allowed 0.4
Clipped gradient with value 0.5 while allowed 0.5
Clipped gradient with value 0.6 while allowed 0.5
Analyzing sampled molecules given pockets at epoch 24...
Validity over 100 molecules: 87.00%
Connectivity over 87 valid molecules: 5.75%
Uniqueness over 5 connected molecules: 100.00%
Novelty over 5 unique connected molecules: 100.00%
Evaluation took 132.43 seconds
/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/openbabel/__init__.py:14: UserWarning: "import openbabel" is deprecated, instead use "from openbabel import openbabel"
  warnings.warn('"import openbabel" is deprecated, instead use "from openbabel import openbabel"')
/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/numpy/core/fromnumeric.py:3504: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/numpy/core/_methods.py:129: RuntimeWarning: invalid value encountered in scalar divide
  ret = ret.dtype.type(ret / rcount)
/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/trainer/connectors/logger_connector/result.py:233: UserWarning: You called `self.log('kl_div_residue_types/val', ...)` in your `validation_epoch_end` but the value needs to be floating point. Converting it to torch.float32.
  warning_cache.warn(
Traceback (most recent call last):
  File "/bme002.mnt/project/bme-lrossen-20192236/Master_Thesis/DiffSBDD/train.py", line 122, in <module>
    trainer.fit(model=pl_module, ckpt_path=ckpt_path)
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 603, in fit
    call._call_and_handle_interrupt(
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/trainer/call.py", line 38, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 645, in _fit_impl
    self._run(model, ckpt_path=self.ckpt_path)
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 1098, in _run
    results = self._run_stage()
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 1177, in _run_stage
    self._run_train()
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 1200, in _run_train
    self.fit_loop.run()
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/loops/loop.py", line 199, in run
    self.advance(*args, **kwargs)
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/loops/fit_loop.py", line 267, in advance
    self._outputs = self.epoch_loop.run(self._data_fetcher)
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/loops/loop.py", line 200, in run
    self.on_advance_end()
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/loops/epoch/training_epoch_loop.py", line 251, in on_advance_end
    self._run_validation()
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/loops/epoch/training_epoch_loop.py", line 310, in _run_validation
    self.val_loop.run()
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/loops/loop.py", line 206, in run
    output = self.on_run_end()
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/loops/dataloader/evaluation_loop.py", line 180, in on_run_end
    self._evaluation_epoch_end(self._outputs)
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/loops/dataloader/evaluation_loop.py", line 288, in _evaluation_epoch_end
    self.trainer._call_lightning_module_hook(hook_name, output_or_outputs)
  File "/home/20192236/.conda/envs/DiffSBDD/lib/python3.10/site-packages/pytorch_lightning/trainer/trainer.py", line 1342, in _call_lightning_module_hook
    output = fn(*args, **kwargs)
  File "/bme002.mnt/project/bme-lrossen-20192236/Master_Thesis/DiffSBDD/lightning_modules.py", line 402, in validation_epoch_end
    getattr(self, 'sample_and_save' + suffix)(
  File "/bme002.mnt/project/bme-lrossen-20192236/Master_Thesis/DiffSBDD/lightning_modules.py", line 607, in sample_and_save_given_pocket
    visualize(str(outdir), dataset_info=self.dataset_info, wandb=None)
  File "/bme002.mnt/project/bme-lrossen-20192236/Master_Thesis/DiffSBDD/analysis/visualization.py", line 332, in visualize
    plot_data3d(positions, atom_type, dataset_info=dataset_info,
  File "/bme002.mnt/project/bme-lrossen-20192236/Master_Thesis/DiffSBDD/analysis/visualization.py", line 177, in plot_data3d
    ax.w_xaxis.line.set_color("black")
AttributeError: 'Axes3D' object has no attribute 'w_xaxis'